README.md

```diff
@@ -37,185 +41,198 @@ Simple parsing for node.js.
+- Describe your grammar in the nearley syntax. `grammar.ne`:
 
+```js
+main -> (statement "\n"):+
+statement -> "foo" | "bar"
 ```
-$ npm install nearley
-$ node
-> var nearley = require("nearley");
-> var grammar = require("./my-generated-grammar.js");
+
+Check out the wonderful [nearley playground](https://omrelli.ug/nearley-playground/) to explore nearley interactively in your browser.
+
+- Compile the grammar to JS:
+
+```bash
+nearleyc grammar.ne -o grammar.js
 ```
 
-Alternatively, to use a generated grammar in a browser runtime, include the
-`nearley.js` file in a `<script>` tag.
+Add a script to `scripts` in `package.json` that runs the command above if you only have a locally installed copy of nearley.
+
+- Create a parser and feed it data:
+
+```js
+const nearley = require("nearley");
+const grammar = require("./grammar");
+
+// Create a Parser object from our grammar.
+const parser = new nearley.Parser(nearley.Grammar.fromCompiled(grammar));
+
+// Parse something.
+parser.feed("foo\n");
 
-```html
-<script src="nearley.js"></script>
-<script src="my-generated-grammar.js"></script>
+// parser.results is an array of possible parsings.
+// It's empty if we're unable to parse the input.
+// If the input can be interpreted in multiple ways, parser.results will contain all of them.
+console.log(parser.results); // [[[[ "foo" ],"\n" ]]]
 ```
 
+See below for detailed API and grammar syntax specification.
 
-Parser specification
---------------------
+## CLI
 
-This is a basic overview of nearley syntax and usage. For an advanced
-styleguide, see [this file](how-to-grammar-good.md).
+Use `--help` with any of these commands to see available options.
 
-A parser consists of several *nonterminals*, which are constructions in a
-language. A nonterminal is made up of a series of either other nonterminals or
-strings. In nearley, you define a nonterminal by giving its name and its
-expansions.
+- `nearleyc` compiles grammar files to JavaScript.
+- `nearley-test` lets you quickly test a grammar against some input and see the
+  results. It also lets you explore the internal state of nearley's Earley
+  table, in case you find that interesting.
+- `nearley-unparse` inverts a parser into a generator, allowing you to create
+  random strings that match your grammar.
+- `nearley-railroad` generates pretty railroad diagrams from your parser. This
+  is mainly helpful for creating documentation, as (for example) on json.org.
+
+## Parser specification
+
+Let's explore the building blocks of a nearley parser.
+
+### Terminals, nonterminals, rules
+
+- A *terminal* is a string or a token. E.g. keyword `"if"` is a terminal.
+- A *nonterminal* is a combination of terminals and other nonterminals. E.g. an if statement defined as `"if" condition statement` is a nonteminal.
+- A *rule* (or production rule) is a definition of a nonterminal. E.g. `"if" condition statement` is the rule according to which the if statement nonterminal is parsed.
```

Comment:
Embarrassing story: I started a glossary a while back, and accidentally committed it at some point.

https://github.com/Hardmath123/nearley/blob/master/glossary.md

Some things in it might be wrong, so we should probably remove it at some point. But, it's a good starting-point if we want to, as Tim says, "make sure we get this right".